pydantic/main.py

```diff
@@ -823,9 +811,13 @@ def __delattr__(self, item: str) -> Any:
             try:
                 # Note: self.__pydantic_private__ cannot be None if self.__private_attributes__ has items
                 del self.__pydantic_private__[item]  # type: ignore
+                return
             except KeyError as exc:
                 raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}') from exc
-        elif item in self.model_fields:
+
+        self._check_frozen(item, None)
```

Comment:
I think that's a good question, but I also think since setattr currently ignores it we can ignore that for the sake of this PR and leave it as is (namely, private attributes can be edited or deleted for frozen instances).

I think it _might_ make sense to change that (and include the private attributes in the hash), but I suspect it might be too much of a breaking change, either way let's ignore it for now.

Reply:
Also, feel free to open a separate issue or PR to address that, I'm just less confident that will make sense to merge compared to this.

Reply:
I just realised that this would prevent private attributes from ever being set to anything at all, since they're not set during initialisation :laughing:  So I guess this is fine.